python/ray/_raylet.pyx

```diff
@@ -588,18 +596,20 @@ cdef execute_task(
         # all past signals.
         ray_signal.reset()
 
-    # Reset the state of the worker for the next task to execute.
-    # Increase the task execution counter.
-    worker.function_actor_manager.increase_task_counter(
-        job_id, function_descriptor)
+    if execution_info.max_calls != 0:
```

Comment:
This should just get moved into c++ (doesn't need to be this PR)

Reply:
yeah!